jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -273,12 +273,37 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
 			}
 			if (!dumpType.isEmpty()) {
 				String request = dumpType;
-				if (parts.length == 2) {
-					String fileDirective = ("system".equals(dumpType) && IPC.isZOS) ? ":dsn=" : ":file="; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-					request += fileDirective + parts[1];
+				boolean foundSystemRequest = false;
+				boolean heapDump = "heap".equals(dumpType);
+				boolean systemDump = "system".equals(dumpType);
+				String options = ""; //$NON-NLS-1$
+				String separator = ":"; //$NON-NLS-1$
+				for (int i = 1; i < parts.length; i++) {
+					String option = parts[i];
+					boolean isRequest = option.startsWith("request=");
+					boolean isOpts = option.startsWith("opts=");
+					if (isRequest || isOpts) { //$NON-NLS-1$ //$NON-NLS-2$
+						if (heapDump && isOpts) {
+							// opts= are only valid for heap dumps
+							break;
+						}
+						options += separator + option;
+						if (systemDump && isRequest) {
+							foundSystemRequest = true;
+						}
+					} else {
+						String fileDirective = (systemDump && IPC.isZOS) ? "dsn=" : "file="; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+						options += separator + fileDirective + option;
+					}
+					separator = ","; //$NON-NLS-1$
+				}
+				if (systemDump && !foundSystemRequest) {
+					// default options for a system dump if the user didn't specify
+					options += "request=exclusive+prepwalk";  //$NON-NLS-1$
```

Comment:
I'll set default options for javacore and heap dumps as well.
javacore - `request=exclusive+preempt`
heap - `request=exclusive+prepwalk`

Reply:
As per [1],  by default `-Xdump:java` has `request=exclusive+preempt`, and `-Xdump:heap` has `request=exclusive+compact+prepwalk,`.

[1] https://www.eclipse.org/openj9/docs/xdump/

Reply:
This needs a separator before "request".
Perhaps `options` should be a `StringBuilder`?